deps: replace zeebe-node with @camunda8/sdk

jwulf opened the pull request:
fixes #4109

barmac commented:
We had just a session together. The mocks need to be fixed. @jwulf will continue on the source code migration.

Discovered https://github.com/camunda/camunda-modeler/issues/4236.

barmac commented:
I rebased this on top of `develop` so that you can build it locally.

As a side note, you can also run the application via `npm start` which is significantly faster than waiting for the app to be built. You may need to restart the script once you change code in `app` directory.

barmac commented:
I pushed changes of the test files which fix some of the failing cases. To be continued.

barmac commented:
Let's make it a draft until it's ready for merge.

barmac commented:
@jwulf zeebe-node [uses property `port` to later construct the URL](https://github.com/camunda-community-hub/zeebe-client-node-js/blob/7969ce1808c96a87519cb1a3f279287f30637c4b/src/lib/ConfigurationHydrator.ts#L238), but I can't see a relevant property in the SDK config. Is it smart enough to figure out defaults for HTTP(s) or do we need to add them to the Zeebe address?

barmac commented:
Also it looks like the SDK expects us to pass a path to the custom root certificates (e.g. `CAMUNDA_CUSTOM_ROOT_CERT_PATH`) while in zeebe-node we would pass a Buffer with certificates.

barmac commented:
I've migrated most of the tests to the new way of creating the client. @jwulf please take a look at the comments above so that we can proceed with this PR: https://github.com/camunda/camunda-modeler/pull/4233#issuecomment-2056923639, https://github.com/camunda/camunda-modeler/pull/4233#issuecomment-2056930722

nikku commented:
@barmac Let's ensure that it is clear what else we'd need to get this merged (potentially from the SDK side).

barmac commented:
To get this merged, we need to:
* Clarify the port usage: https://github.com/camunda/camunda-modeler/pull/4233#issuecomment-2056923639
* Have a way to include the OS certificates (either built in the library or provide an option to pass a blob with certificates): https://github.com/camunda/camunda-modeler/pull/4233#issuecomment-2056931596

@jwulf Please let me know if you need any support with this.

jwulf commented:
> To get this merged, we need to:
> 
> * Clarify the port usage: [deps: replace zeebe-node with @camunda8/sdk #4233 (comment)](https://github.com/camunda/camunda-modeler/pull/4233#issuecomment-2056923639)
> * Have a way to include the OS certificates (either built in the library or provide an option to pass a blob with certificates): [deps: replace zeebe-node with @camunda8/sdk #4233 (comment)](https://github.com/camunda/camunda-modeler/pull/4233#issuecomment-2056931596)
> 
> @jwulf Please let me know if you need any support with this.

OK. I have moved the custom certificate management code into the SDK itself. At the moment it can take a file path to a custom certificate. 

I've added tests for the feature, as well as tests on Windows in CI (didn't have any of those). 

I have opened an issue to support passing the certificate itself as a Buffer or String, as zeebe-node used to. 

https://github.com/camunda/camunda-8-js-sdk/issues/142

I'll get to that at the end of this week or beginning of next week.

jwulf commented:
Version 8.5.2 of the Camunda 8 JS SDK supports `CAMUNDA_CUSTOM_ROOT_CERT_STRING` as a configuration parameter. This allows you to supply a custom TLS certificate as a string. 

It will be aggregated with the system certificates automatically, and used to verify the TLS certificate used to secure the Zeebe Gateway.

barmac commented:
Unfortunately, I need to postpone this to the week after CamundaCon.

barmac commented:
I am looking into this today.

barmac commented:
In order to check the SDK configuration, I implemented a connection test in https://github.com/camunda/zeebe-connection-test/pull/12

barmac commented:
@jwulf Is the basic auth supported in the SDK? I couldn't find a configuration option for this even though it was supported in zeebe-node.

barmac commented:
This is blocked by https://github.com/camunda/camunda-8-js-sdk/issues/165

barmac commented:
> This is blocked by [camunda/camunda-8-js-sdk#165](https://github.com/camunda/camunda-8-js-sdk/issues/165)

We are unblocked now 🚀

nikku commented:
@barmac You can verify this works via https://github.com/camunda/zeebe-connection-test/pull/11

barmac commented:
This PR is now in the backlog so we will not work on this right now.

barmac commented:
@abdul99ahad will take this over.

nikku commented:
Superseded by https://github.com/camunda/camunda-modeler/pull/4817.
